### ISSUE TITLE
Upgrade Lerna to ^4

### ISSUE THREAD

**Issue by kriskowal** [endojs/endo#1060]
There are a number of security advisories that are likely to just-go-away™ if we upgrade to Lerna 4. This will require reviewing whether we need to update the patch for lerna-conventional-commits.

**Comment by mhofman** [comment]
This is relevant for `agoric-sdk` as well

**Comment by SMotaal** [comment]
I hit a problem running `yarn` after cloning the repo which I traced to [this line in `node-gyp@5.1.1`](https://github.com/nodejs/node-gyp/blob/748478eb195208b7e30f65c08b2e7d7d684253e0/lib/configure.js#L100):

```js
    var config = process.config || {}
```

I managed to get `yarn` to work but only with `devDependecies: { "lerna": "^5.0.0" }` which installed `node-gyp@9.3.1` as I traced the fix to [this line in `node-gyp@8.2.0`](https://github.com/nodejs/node-gyp/blob/bc47cd60b986eaa55a23050d8f72d1cc117bdba0/lib/configure.js#L101) or newer:

```js
    var config = process.config ? JSON.parse(JSON.stringify(process.config)) : {}
```

I suppose it may be necessary to skip `^4` or reconsidering `lerna` all together if `yarn` could be sufficient to meet the requirements, which I would not know myself at this point but could explore.

---

<details><summary>Boring stuff</summary>

To trace dependency graphs I use:

```sh
npm list node-gyp
```

**Note**: The top-level package for endojs/endo still defines `{ "name": "SES" }` which is used below.

When I traced with `devDependecies: { "lerna": "^4.0.0" }` in `SES` I got this graph:

```log
SES@ /Users/daflair/Projects/Agoric/endojs/endo-fix-types-missing-docs
└─┬ lerna@4.0.0
  ├─┬ @lerna/add@4.0.0
  │ └─┬ pacote@11.3.5
  │   └─┬ @npmcli/run-script@1.8.6
  │     └── node-gyp@7.1.2
  └─┬ @lerna/bootstrap@4.0.0
    └─┬ @lerna/run-lifecycle@4.0.0
      └─┬ npm-lifecycle@3.1.5
        └── node-gyp@5.1.1
```

When I traced with `devDependecies: { "lerna": "^5.0.0" }` in `SES` I got this graph:

```log
SES@ /Users/daflair/Projects/Agoric/endojs/endo-fix-types-missing-docs
└─┬ lerna@5.6.2
  └─┬ @lerna/add@5.6.2
    └─┬ pacote@13.6.2
      └─┬ @npmcli/run-script@4.2.1
        └── node-gyp@9.3.1
```

The goal here was to get `yarn` to execute without errors.

Here is the `yarn` log in case it is useful:

```log
yarn install v1.22.0
[1/5] 🔍  Validating package.json...
[2/5] 🔍  Resolving packages...
warning lerna > @lerna/bootstrap > @npmcli/arborist > @npmcli/move-file@2.0.1: This functionality has been moved to @npmcli/fs
warning lerna > @lerna/add > pacote > cacache > @npmcli/move-file@2.0.1: This functionality has been moved to @npmcli/fs
warning lerna > @lerna/bootstrap > @npmcli/arborist > readdir-scoped-modules@1.1.0: This functionality has been moved to @npmcli/fs
[3/5] 🚚  Fetching packages...
[4/5] 🔗  Linking dependencies...
warning " > eslint-plugin-prettier@3.4.1" has unmet peer dependency "prettier@>=1.13.0".
warning "workspace-aggregator-5d7ac561-0d1f-4417-9605-13a1e4a4f5ae > ses-integration-test > @rollup/plugin-commonjs@13.0.2" has incorrect peer dependency "rollup@^2.3.4".
[5/5] 🔨  Building fresh packages...
success Saved lockfile.
$ patch-package
patch-package 6.4.7
Applying patches...
@lerna/conventional-commits@3.22.0 ✔
acorn@7.4.1 ✔

Warning: patch-package detected a patch file version mismatch

  Don't worry! This is probably fine. The patch was still applied
  successfully. Here's the deets:

  Patch file created for

    @lerna/conventional-commits@3.22.0

  applied to

    @lerna/conventional-commits@5.6.2
  
  At path
  
    node_modules/@lerna/conventional-commits

  This warning is just to give you a heads-up. There is a small chance of
  breakage even though the patch was applied successfully. Make sure the package
  still behaves like you expect (you wrote tests, right?) and then run

    patch-package @lerna/conventional-commits

  to update the version in the patch file name and make this warning go away.

---
patch-package finished with 1 warning(s).
✨  Done in 48.76s.
```

Here is the relevant `yarn.lock` entry for reference:

https://github.com/endojs/endo/blob/a1165343d19454cb6be74722d07e7ed02d567c09/yarn.lock#L8620-L8621

</details>

**Comment by turadg** [comment]
The relevant agoric-sdk issue is https://github.com/Agoric/agoric-sdk/issues/4660

@SMotaal if you want to try executing either of the designs in that ticket (on either repo), I would review the PR promptly.

**Comment by SMotaal** [comment]
This issue was addressed with #1415 and #1418 🎉

**Comment by turadg** [comment]
Thanks @SMotaal !
